### ISSUE TITLE
Add Cut, Copy, and Paste items to Edit menu

### ISSUE THREAD

**Issue by justinclift** [sqlitebrowser/sqlitebrowser#710]
#### I'm opening this issue because:

- [ ] DB4S is crashing
- [ ] DB4S has a bug
- [x] DB4S needs a feature
- [ ] DB4S has another problem: ___

### Details for the issue

We have an Edit menu, but the standard "Cut, Copy, Paste" items are missing from it.

When our Copy/Paste code is in good shape, we should probably add them back in.

**Comment by MKleusberg** [comment]
I'm not 100% sure about this. I think the Edit menu is probably the wrong place for this because a Copy and a Paste action could be quite confusing there because of our tabbed window layout. Depending on the currently active tab we'd have to change the menu actions and even then this would be difficult to get your head around.

So I'd propose to not add them to the Edit menu but instead add a context menu that opens when right clicking. This way it's much clearer what data the menu actions operate on in a given context. As a matter of fact I've just added that context menu :wink:

@justinclift What do you think? Should we leave it at that and close the issue?

**Comment by justinclift** [comment]
Ahhh, yeah I see what you mean.

How hard would it be for us to dynamically add/remove them from the Edit menu when the user goes to, or changes away from, the Browse Data tab?

Hmmm... ideally, they'd probably show up for both the Browse Data and Execute SQL tabs.  They'd both do the same general thing (concept wise), but we'd need to fire off different functions for each. 

The ExecuteSQL one would probably be fairly simple... we already have copy&paste functionality in that from QScintilla don't we?

For the Browse Data tab, that'd fire off the cut&paste code @innermous wrote.  Still, if it's not a large amount of effort we should probably give it a go.  These menu items are pretty much "standard functionality" which users expect to have available in that menu... and we do have a wide range of users out there (from newbies to expert, etc).

**Comment by MKleusberg** [comment]
True. But one more concern :wink: when I'm in the Execute SQL tab, I might expect the Copy & Paste actions to work on the SQL editor (as you wrote) or on the result view (e.g. for copying the result data to Excel or whatever). To me it just feels like our particular UI design doesn't really fit that well with the Edit menu. But yeah, we could just hide the Copy & Paste items for all tabs except the Browse Data tab.
